### PR TITLE
[Content Fragment List] Display in a specific order

### LINKED ISSUE #698: [Content Fragment List] Display in a specific order
## Bug Report

**Current Behavior**
When author configured content fragment list, Content fragments displayed in desc order of lastModified. Whenever the author updated the content fragments inside the content fragment list this display order will change. 

**Expected behavior/code**
When the author updates the content fragment inside the content fragment list. Display order should not change it should remain the same. 

**Environment**

**Possible Solution**
To provide a consistent display order, Have to add orderby predicate in query builder that generates content fragment in ContentFragmentListImpl.java.

### PR DESCRIPTION
**Pull request by bpauli**
- add order by and sort order drop down to cfm list component
- enhance data source servlet to populate list for order by drop down
- enhance cfm list model to create additional predicate if order by and sort oder property is set
- adapt unit tests

fixes #698

### CODE REVIEW

File: content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/_cq_dialog/.content.xml [hunk 1]
```diff
@@ -54,6 +54,38 @@
                                         fieldLabel="Tags"
                                         multiple="{Boolean}true"
                                         name="./tagNames"/>
+                                    <orderBy
+                                        jcr:primaryType="nt:unstructured"
+                                        sling:resourceType="granite/ui/components/coral/foundation/form/select"
+                                        emptyOption="{Boolean}true"
+                                        fieldDescription="Select a field the list is ordered by"
```
Comment (richardhand) [inline content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/_cq_dialog/.content.xml]: "Select a field to order the list by."

File: content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/_cq_dialog/.content.xml [hunk 2]
```diff
@@ -54,6 +54,38 @@
...
+                                            <asc
+                                                jcr:primaryType="nt:unstructured"
+                                                text="Ascending"
+                                                value="asc"/>
+                                            <desc
+                                                jcr:primaryType="nt:unstructured"
+                                                text="Descending"
+                                                value="desc"/>
+                                        </items>
+                                    </sortOrder>e
```
Comment (richardhand) [inline content/src/content/jcr_root/apps/core/wcm/components/contentfragmentlist/v1/contentfragmentlist/_cq_dialog/.content.xml]: "</sortOrder>" (remove 'e')

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/ModelElementsDataSourceServlet.java [hunk 3]
```diff
@@ -73,6 +74,8 @@ protected void doGet(@NotNull SlingHttpServletRequest request, @NotNull SlingHtt
         // First try to get the model path from request parameters
         // otherwise determine model path from component resource.
         RequestParameter modelPathRequestParameter = request.getRequestParameter(PARAMETER_AND_PN_MODEL_PATH);
+        String resourcePath = request.getRequestPathInfo().getResourcePath();
+        boolean isOrderBy = StringUtils.endsWith(resourcePath, "/orderBy");
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/ModelElementsDataSourceServlet.java]: This requires the node consuming the datasource to be named `orderBy` if I read this correct? We could make this more flexible by allowing an additional property to be supplied to the datasource.

File: bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/ModelElementsDataSourceServlet.java [hunk 4]
```diff
@@ -49,13 +49,15 @@
...
                 "sling.servlet.resourceTypes=" + ModelElementsDataSourceServlet.RESOURCE_TYPE,
+                "sling.servlet.resourceTypes=" + ModelElementsDataSourceServlet.RESOURCE_TYPE_ORDER_BY,
                 "sling.servlet.methods=GET",
                 "sling.servlet.extensions=html"
         }
 )
 public class ModelElementsDataSourceServlet extends AbstractDataSourceServlet {
 
     public static final String RESOURCE_TYPE = "core/wcm/components/contentfragmentlist/v1/datasource/elements";
+    public static final String RESOURCE_TYPE_ORDER_BY = "core/wcm/components/contentfragmentlist/v1/datasource/elements";
```
Comment (richardhand) [inline bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/servlets/contentfragment/ModelElementsDataSourceServlet.java]: -> "core/wcm/components/contentfragmentlist/v1/datasource/orderby"

**Comment by richardhand** [comment]
@bpauli  - how about the feedback for a default option in the orderby select?

**Comment by bpauli** [comment]
@richardhand The default order by and sort by is added in the latest commit https://github.com/adobe/aem-core-wcm-components/pull/739/commits/0464c8d6474162ad196892b04f774e9d32b9abfd

**Pull request merged**